### PR TITLE
Feat/wait for digest

### PR DESCRIPTION
**Pull request by 0xArdi**
## Proposed changes

This PR adds support for checking if the receipt is not None when sending a solana tx.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: plugins/aea-ledger-solana/aea_ledger_solana/solana.py [hunk 1]
```diff
@@ -429,14 +429,16 @@ def _try_send_signed_transaction(
             )
         stxn = self.deserialize_tx(tx=tx_signed)
         txn_resp = self._api.send_raw_transaction(bytes(stxn.serialize()))
-        retries = 2
+        retries = 15
```
Comment (0xArdi) [inline plugins/aea-ledger-solana/aea_ledger_solana/solana.py:432]: 2 seconds was too little in my experience

Comment (DavidMinarsch) [reply plugins/aea-ledger-solana/aea_ledger_solana/solana.py:432]: Should this not be a method parameter (can be default)

Comment (angrybayblade) [reply plugins/aea-ledger-solana/aea_ledger_solana/solana.py:432]: Actually, the `_try_send_signed_transaction` method is implementation is wrong. It's only supposed to send the transaction and return the tx digest, but instead it waits for the transaction to settle and returns the tx receipt. I suggest we fix this in the next iteration, so the settlement can be handled by the skill not the ledger.

**Review by angrybayblade — DISMISSED** [review]

File: plugins/aea-ledger-solana/aea_ledger_solana/solana.py (hunk 1 above)
Comment (angrybayblade) [reply plugins/aea-ledger-solana/aea_ledger_solana/solana.py:432]: @0xArdi Can you take a look at this PR https://github.com/valory-xyz/open-aea/pull/710

Comment (0xArdi) [reply plugins/aea-ledger-solana/aea_ledger_solana/solana.py:432]: I suggest we also add the suggestion by @DavidMinarsch on the next release.

**Review by angrybayblade — APPROVED** [review]

**Pull request merged**